mypy/modulefinder.py

```diff
@@ -443,10 +444,23 @@ def find_modules_recursive(self, module: str) -> List[BuildSource]:
         names = sorted(self.fscache.listdir(package_path))
         for name in names:
             # Skip certain names altogether
-            if name == '__pycache__' or name.startswith('.') or name.endswith('~'):
+            if (
+                name in ("__pycache__", "site-packages", "node_modules")
+                or name.startswith(".")
+                or name.endswith("~")
+            ):
```

Comment:
Do we still need this once we have the regular expression? Or if the default exclusions are always enabled, this can be left and the code path below is only taken if there are non-default exclusions.

It may be worth having this code in a utility function that gets called from here and `find_sources.py` to avoid duplication. Already the implementations are slightly different, and it's easy to forget to update one of them.